### ISSUE TITLE
I can't deploy with Next v12.3.1 and @sls-next/serverless-component v3.7.0. CloudFront returns 503

### ISSUE THREAD

**Issue by TxusBlack** [serverless-nextjs/serverless-next.js#2498]
Hi everyone! Before we're using `"next": "^12.2.5"` but actually we need to use `"next": "^12.3.1",`, but when we deploy, the API returns an 503 error. I attach a picture of the error:

![image](https://user-images.githubusercontent.com/13071556/198078732-b4cdc439-4bb2-44ee-b0de-8745f5b6483b.png)

I also see the Cloudwatch Log and we have an error inside the library:

![image](https://user-images.githubusercontent.com/13071556/198078881-c92f6964-6592-4f06-9113-e990ce6290e8.png)

We're applying this solution, but how I explain, we need the version 12.3.1 for Next.js.

This is my serverless.yaml:

```
name: ...

nextApp:
  component: "@sls-next/serverless-component"
  inputs:
    # nextConfigDir: ../../
    aliases: ["..."]
    bucketName: "..."
    bucketRegion: "..."
    certificateArn: ...
    name:
      imageLambda: "..."
      defaultLambda: "..."
      apiLambda: "..."
    description: "..."
    runtime: "nodejs16.x"
    cloudfront:
      distributionId: "..."
      comment: '...'
    memory: 2048
    timeout: 30
    removeOldLambdaVersions: true
    build:
      args: ["build"]
      env:
        NEXT_PUBLIC_ENV: "production"
        AWS_ACCESS_KEY_ID: "..."
        AWS_SECRET_ACCESS_KEY: "..."
```

And for deploy I'm using:
`npx serverless@2.72.2`

Thank you!

- OS/Environment:
- @sls-next/serverless-component version: ^3.7.0
- Next.js version: ^12.3.1
- MacOS.

**Comment by OscarBC** [comment]
Hello everyone, [I've made a PR to 'iron-sesion' package](https://github.com/vvo/iron-session/pull/558) that also was giving me a 503 error with that exact message. @TxusBlack can you confirm that you are using the iron-session package?

I believe somewhere in the serverless nextjs core the http request/response object is being set in to something that is not a ServerResponse but a simple js object ( {} ). But who knows, maybe the issue is on the iron-session side. 

I'm attaching a screenshot of the console log of the contents of the "response" and "response.headers" values. 

Where it's clear that the response headers object is just "not null". 

![Captura de Pantalla 2022-11-04 a la(s) 1 16 00](https://user-images.githubusercontent.com/5251988/199925146-e2f68796-335c-4cac-beb3-a9c3790dd118.png)
In blue is the "typeof" of the res.headers object. I believe it should be an HttpResponseObject and not a "simple js object"

**Comment by OscarBC** [comment]
> 

I wanted to say that I have a hunch that the way the headers object is set that it might have something to do with why the middleware aren't working (But I don't really have clue and just wanted to bring it to the table)

**Comment by vizardkill** [comment]
If indeed the error occurs when using the iron-session package

**Comment by ykurniawan** [comment]
What version of iron-session are you using? I managed to fix by downgrade it to "iron-session": "6.1.3"

**Comment by TxusBlack** [comment]
Hieveryone! The solution we used was to downgrade iron-session and now it's working, so it would be fine to use the latest version for some specific features we need.
